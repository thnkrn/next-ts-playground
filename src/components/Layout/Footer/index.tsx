import { styled } from '@compiled/react'
import Icon from 'components/Common/Icon'
import { CONTACT_LIST } from 'constants/contact'
import TYPO from 'constants/typography'

const StyledFooter = styled.footer`
  color: white;
  margin-top: 80px;
  text-align: center;
  border-top: 1px solid #fff;
  padding: 20px;
`

const StyledContactWrapper = styled.div`
  @media screen and (max-width: 480px) {
    flex-direction: column;
    margin-top: 20px;
  }

  display: flex;
  justify-content: center;
  margin-top: 40px;
`

const StyledContactList = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;

  span {
    margin: 3px 0 0 10px;
  }

  &:not(:first-child) {
    @media screen and (max-width: 480px) {
      margin: 20px 0 0 !important;
    }

    margin-left: 50px;
  }
`

const Footer = () => (
  <StyledFooter>
    <StyledContactWrapper>
      {CONTACT_LIST.map((list) => (
        <StyledContactList key={list.title}>
          <Icon src={list.icon} />
          {list.isLink ? (
            <span className={TYPO.contact}>
              <a href={list.link}>{list.title}</a>
            </span>
          ) : (
            <span className={TYPO.contact}>{list.title}</span>
          )}
        </StyledContactList>
      ))}
    </StyledContactWrapper>
  </StyledFooter>
)

export default Footer
