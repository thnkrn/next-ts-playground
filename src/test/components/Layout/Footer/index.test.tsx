import Footer from 'components/Layout/Footer'
import { render } from 'test/util'

describe('Footer', () => {
  describe('Rendering', () => {
    it('should match snapshot', () => {
      const { asFragment } = render(<Footer />)

      expect(asFragment()).toMatchSnapshot()
    })
  })
})
