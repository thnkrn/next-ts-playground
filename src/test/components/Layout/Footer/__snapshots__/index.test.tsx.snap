// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Rendering should match snapshot 1`] = `
<DocumentFragment>
  <style
    data-cmpld="true"
  >
    ._syaz1x77{color:white}._19pku29b{margin-top:80px}._y3gn1h6o{text-align:center}._x3doxja1{border-top:1px solid #fff}._ca0qgktf{padding-top:20px}._u5f3gktf{padding-right:20px}._n3tdgktf{padding-bottom:20px}._19bvgktf{padding-left:20px}@media screen and (max-width:480px){._za9y1wqb{margin-top:60px}}
  </style>
  <footer
    class="_za9y1wqb _syaz1x77 _19pku29b _y3gn1h6o _x3doxja1 _ca0qgktf _u5f3gktf _n3tdgktf _19bvgktf"
  >
    <style
      data-cmpld="true"
    >
      ._1e0c1txw{display:flex}._1bah1h6o{justify-content:center}._19pk1ylp{margin-top:40px}@media screen and (max-width:480px){._46cc1bp4{flex-direction:column}._za9y19bv{margin-top:10px}}
    </style>
    <div
      class="_46cc1bp4 _za9y19bv _1e0c1txw _1bah1h6o _19pk1ylp"
    >
      <style
        data-cmpld="true"
      >
        ._4cvr1h6o{align-items:center}._1ipk1l7b span{margin-top:3px}._1kdcidpf span{margin-right:0}._3glyidpf span{margin-bottom:0}._1w2u19bv span{margin-left:10px}._19qr12am:not(:first-child){margin-left:50px}@media screen and (max-width:480px){._kv0z1txs:not(:first-child){margin-top:20px!important}._urhrgrf3:not(:first-child){margin-right:0!important}._17kpgrf3:not(:first-child){margin-bottom:0!important}._1ol6grf3:not(:first-child){margin-left:0!important}}
      </style>
      <div
        class="_1e0c1txw _4cvr1h6o _1bah1h6o _1ipk1l7b _1kdcidpf _3glyidpf _1w2u19bv _19qr12am _kv0z1txs _urhrgrf3 _17kpgrf3 _1ol6grf3"
      >
        <svg
          class="_1bsb1tcg _4t3i1tcg _tzy4idpf _18t5kb7n icon loaded"
        >
          <use
            href="/icons/icons-sprite.svg#email"
          />
        </svg>
        <span
          class="fancy"
        >
          t.ariyagusolsuthi@gmail.com
        </span>
      </div>
      <div
        class="_1e0c1txw _4cvr1h6o _1bah1h6o _1ipk1l7b _1kdcidpf _3glyidpf _1w2u19bv _19qr12am _kv0z1txs _urhrgrf3 _17kpgrf3 _1ol6grf3"
      >
        <svg
          class="_1bsb1tcg _4t3i1tcg _tzy4idpf _18t5kb7n icon loaded"
        >
          <use
            href="/icons/icons-sprite.svg#github"
          />
        </svg>
        <span
          class="fancy"
        >
          <a
            href="https://github.com/thnkrn"
          >
            Github
          </a>
        </span>
      </div>
      <div
        class="_1e0c1txw _4cvr1h6o _1bah1h6o _1ipk1l7b _1kdcidpf _3glyidpf _1w2u19bv _19qr12am _kv0z1txs _urhrgrf3 _17kpgrf3 _1ol6grf3"
      >
        <svg
          class="_1bsb1tcg _4t3i1tcg _tzy4idpf _18t5kb7n icon loaded"
        >
          <use
            href="/icons/icons-sprite.svg#linkedin"
          />
        </svg>
        <span
          class="fancy"
        >
          <a
            href="https://www.linkedin.com/in/thanakorn-ariyagusolsuthi"
          >
            Linkedin
          </a>
        </span>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
